src/background/get-persisted-data.ts

```diff
@@ -63,12 +88,35 @@ export function getPersistedData(
         return Promise.resolve(persistedData); //empty object
     }
 
-    const promises: Array<Promise<any>> = dataKeysToFetch.map(key => {
-        return indexedDBInstance.getItem(key).then(data => {
-            const persistedDataKey = keyToPersistedDataMappingOverrides[key] ?? `${key}Data`;
-            persistedData[persistedDataKey] = data;
+    const promises = getGlobalPersistedPromises(dataKeysToFetch, indexedDBInstance, persistedData);
+
+    return Promise.all(promises).then(() => persistedData);
+}
+
+export async function getAllPersistedData(indexedDBInstance: IndexedDBAPI): Promise<PersistedData> {
+    const persistedData = { tabData: {} } as PersistedData;
+
+    let promises = getGlobalPersistedPromises(
+        IndexedDBDataKeys.globalKeys,
+        indexedDBInstance,
+        persistedData,
+    );
+
+    const knownTabIds: number[] = await indexedDBInstance.getItem(IndexedDBDataKeys.knownTabIds);
+    if (knownTabIds && knownTabIds.length > 0) {
```

Comment:
This bit of code is doing a lot of heavy lifting. It might be good to break it out into a couple separate pure functions that we can call here to make it more readable and test individually.

Reply:
Thanks Madalyn! I've pushed a merge to make sure this doesn't break the service-init you added yesterday and also separated out into functions. Do you think it's necessary to test all the helper functions separately? I'd prefer to not export them and I'm not sure if there's a way to test without exporting.

Reply:
this looks great actually! there are ways to test non-exported functions by creating a "testable" version of the class inside the test file that extends the original class and has methods that act as handles to the internals but it's not blocking.